types/schemas/discriminated/index.ts

```diff
@@ -0,0 +1,33 @@
+import {PrimaryApplicationTypes} from '../digitalPlanningApplication/enums/ApplicationTypes';
+import {User} from './User';
+
+interface GenericApplication<T extends PrimaryApplicationTypes> {
+  primaryApplicationType: T;
```

Comment:
Not totally sold on the name for this property, but it is required to be above or at an equal hierarchy level of anywhere `T` is passed in.

This means if we need variants based on `T` outside of `data.application` (I think we do?), we can't just keep this in `data.application.applicationType`.